### PR TITLE
Added GOTO window!

### PR DESCRIPTION
**Pull request by 2020karaulov2020**
https://github.com/wootguy/bspguy/issues/13

### CODE REVIEW

**Comment by 2020karaulov2020** [comment]
@wootguy this is correct or in bspguy coordinates are different from Valve Hammer Editor?

File: src/editor/Gui.cpp [hunk 1]
```diff
@@ -1631,6 +1638,53 @@ void Gui::drawKeyvalueEditor_RawEditTab(Entity* ent) {
...
+		}
+		ImGui::Text("Coordinates");
+		ImGui::PushItemWidth(inputWidth);
+		ImGui::DragFloat("##xpos", &coordinates.x, 0.1f, 0, 0, "X: %f");
+		ImGui::SameLine();
+		ImGui::DragFloat("##ypos", &coordinates.y, 0.1f, 0, 0, "Y: %f");
+		ImGui::SameLine();
+		ImGui::DragFloat("##zpos", &coordinates.z, 0.1f, 0, 0, "Z: %f");
+		ImGui::PopItemWidth();
+		ImGui::Text("Views");
```
Comment (wootguy) [inline src/editor/Gui.cpp:1664]: rename to "Angles". Mappers/ripenters will understand that better.

File: src/editor/Gui.cpp [hunk 2]
```diff
@@ -1631,6 +1638,53 @@ void Gui::drawKeyvalueEditor_RawEditTab(Entity* ent) {
...
+		ImGui::PushItemWidth(inputWidth);
+		ImGui::DragFloat("##xangles", &views.x, 0.1f, 0, 0, "X: %f");
+		ImGui::SameLine();
+		ImGui::DragFloat("##yangles", &views.y, 0.1f, 0, 0, "Y: %f");
+		ImGui::SameLine();
```
Comment (wootguy) [inline src/editor/Gui.cpp:1669]: The Y value should be removed since camera rolling isn't supported, and it messes up the controls when updated anyway.

**Review by wootguy — CHANGES_REQUESTED** [review]
Looks good, and the coordinates are correct. I'm not sure about "Views". Y/Z values might be swapped/inverted in-game, but would need to check.

File: src/editor/Gui.cpp [hunk 3]
```diff
@@ -1631,6 +1638,53 @@ void Gui::drawKeyvalueEditor_RawEditTab(Entity* ent) {
...
+		float inputWidth = (ImGui::GetWindowWidth() - (padding + style.ScrollbarSize)) * 0.33f;
+		if (showGOTOWidget_update)
+		{
+			coordinates = app->cameraOrigin;
+			views = app->cameraAngles;
+			showGOTOWidget_update = false;
+		}
+		ImGui::Text("Coordinates");
+		ImGui::PushItemWidth(inputWidth);
+		ImGui::DragFloat("##xpos", &coordinates.x, 0.1f, 0, 0, "X: %f");
```
Comment (wootguy) [inline src/editor/Gui.cpp:1658]: I don't think pinpoint accuracy matters much for this feature. Maybe use `%.0f` instead of `%f` for these values.

**Comment by 2020karaulov2020** [comment]
Now problems fixed?

File: src/editor/Gui.cpp (hunk 1 above)
Comment (2020karaulov2020) [reply src/editor/Gui.cpp:1665]: Ok

File: src/editor/Gui.cpp (hunk 2 above)
Comment (2020karaulov2020) [reply src/editor/Gui.cpp]: Ok

File: src/editor/Gui.cpp (hunk 3 above)
Comment (2020karaulov2020) [reply src/editor/Gui.cpp]: Ok

File: src/editor/Gui.cpp (hunk 1 above)
Comment (wootguy) [reply src/editor/Gui.cpp:1665]: the button got renamed to "Angles" instead of this "Views" text. "GOTO" makes sense for the button.

Comment (2020karaulov2020) [reply src/editor/Gui.cpp]: Yes)) Now fixed ?

**Comment by 2020karaulov2020** [comment]
Yes)) Now fixed ?

**Comment by wootguy** [comment]
Now fixed!

**Pull request merged**